### DISCUSSION TITLE
UMEP standalone functionalities possible?

### DISCUSSION THREAD

**Discussion by Witellsteim** [UMEP-dev/UMEP#522] [Q&A]
Hello all,

First let me say, that i really enjoy working with the UMEP plugins. Recently i was looking into the python scripts for several modules since i was wondering, how i could create a qgis independent standalone script to call on some of the UMEP functionalities. For me as a bit of a newcomer to python (not to programming in general though) this seemed difficult, since many classes are depending and inheriting from qgis objects, and i am not familiar with those. However i think there might be an "easyish" way that i am missing.
Maybe some of you guys might be able to send me on the right way?

Best regards and thank you,
W.

**Comment by nilswallenberg** [comment]
Hi,

Thank you! :)

My suggestions is that you check out the PyQGIS Developer Cookbook: https://docs.qgis.org/3.28/en/docs/pyqgis_developer_cookbook/index.html

There's even a section on Developing Python Plugins: https://docs.qgis.org/3.28/en/docs/pyqgis_developer_cookbook/plugins/index.html

Cheers,
Nils

**Comment by Witellsteim** [comment]
Thanks for the quick answer, i'll definitely have a look into the cookbook. I also found the page for UMEP for processing (https://umep-docs.readthedocs.io/en/latest/UMEPforProcessing.html). Do i understand correctly that this provides a way to access UMEP algorithms via a standalone script?
Best regards,
W.

**Reply by nilswallenberg** [reply]
Yes, that is correct. A standalone script that you can run outside of QGIS.
